### ISSUE TITLE
How can I write async?

### ISSUE THREAD

**Issue by GF-Huang** [influxdata/influxdb-client-csharp#87]
I only see the sync methods.

![image](https://user-images.githubusercontent.com/4510984/80312241-bc6adf00-8816-11ea-8478-b6397f81a817.png)

**Comment by bednar** [comment]
Hi @GF-Huang,

The `WriteApi` writes data in batches in a background thread. 

Regards

**Comment by GF-Huang** [comment]
So how can I wait it finish in async?

**Comment by bednar** [comment]
You could do something like that:

```csharp
_writeApi = Client.GetWriteApi();
_writeApi.WriteRecord(_bucket.Name, _organization.Id, WritePrecision.Ns, "temperature,location=north value=70.0 2");

var finishWrite = Task.Run(() => _writeApi.Dispose());
await finishWrite;
```

**Comment by GF-Huang** [comment]
Why not add such as `WriteAsync`, `BatchWriteAsync` or implementing the [`IAsyncDisposable`](https://docs.microsoft.com/en-us/dotnet/api/system.iasyncdisposable?view=netcore-3.1) interface?

**Comment by bednar** [comment]
@GF-Huang, thanks for that.

I think that `IAsyncDisposable` is a right way and we will add support for that in a next release.

Thanks again

**Comment by GF-Huang** [comment]
Welcome, I hope influxdb become better.

**Comment by bednar** [comment]
Hi @GF-Huang, I detaily looked at `IAsyncDisposable`. Our resources that we have to dispose doesn't support async disposing and implementation such as:

```c#
public async ValueTask DisposeAsync()
{
    await Task.Run(() => Dispose());
}
```

is considered as a bad practice :(. So I will have to do a more research how to improve our `WriteApi` to support asynchronous disposing.

- https://www.infoq.com/articles/Async-API-Design/

**Comment by GF-Huang** [comment]
I guess you may have a problem that the background thread has difficulty notifying the main thread that the task has completed, if so, did you have consider [`TaskCompletionSource`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcompletionsource-1?view=netcore-3.1) for notify caller thread that the background thread already finish?

**Comment by bednar** [comment]
Hi @GF-Huang,

we just implemented [WriteApiAsync](https://github.com/influxdata/influxdb-client-csharp/blob/master/Client/WriteApiAsync.cs) which is simplified version of WriteApi without batching support.

Does it satisfy your requirements?

Regards

**Comment by GF-Huang** [comment]
@bednar The interface design looks great, thank you!

**Comment by bednar** [comment]
Closed by #23
